Bottom: 30f3585a4ccb8fcf229203b162627df41cee733f
Top:    1eb9c73b38589247a57036a7c531e1738883bb1a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 00:09:07 +0100

Refresh of w-c-split-gathering-routines-f-0

---

diff --git a/src/draw.c b/src/draw.c
index efdf499..1c03dc1 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1486,40 +1486,16 @@ DrawElementName (ElementTypePtr Element)
 void
 DrawElementPackage (ElementTypePtr Element)
 {
-  if (Gathering)
-    {
-      ELEMENTLINE_LOOP (Element);
-      {
-        AddPart (line);
-      }
-      END_LOOP;
-      ARC_LOOP (Element);
-      {
-        AddPart (arc);
-      }
-      END_LOOP;
-      return;
-    }
-
-  /* set color and draw lines, arcs, text and pins */
-  if (doing_pinout || doing_assy)
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else if (TEST_FLAG (SELECTEDFLAG, Element))
-    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
-  else if (FRONT (Element))
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else
-    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
+  assert (Gathering);
 
-  /* draw lines, arcs, text and pins */
   ELEMENTLINE_LOOP (Element);
   {
-    _draw_line (line);
+    AddPart (line);
   }
   END_LOOP;
   ARC_LOOP (Element);
   {
-    _draw_arc (arc);
+    AddPart (arc);
   }
   END_LOOP;
 }
